app/src/main/res/layout/activity_track_filters.xml

```diff
@@ -1,9 +1,61 @@
 <?xml version="1.0" encoding="utf-8"?>
-<android.support.v7.widget.RecyclerView
+
+<FrameLayout
   xmlns:android="http://schemas.android.com/apk/res/android"
-  android:id="@+id/trackFiltersList"
+  xmlns:app="http://schemas.android.com/apk/res-auto"
+  xmlns:tools="http://schemas.android.com/tools"
+  android:id="@+id/backgroundDim"
+  style="@style/TrackFilters.BackgroundDim"
   android:layout_width="match_parent"
-  android:layout_height="match_parent"
-  android:paddingLeft="@dimen/track_filters_item_padding_horizontal"
-  android:paddingRight="@dimen/track_filters_item_padding_horizontal"
-  android:clipToPadding="false"/>
+  android:layout_height="match_parent">
+
+  <!-- This is not accessible to the keyboard by design, it should only stop taps from going through it -->
+  <android.support.constraint.ConstraintLayout
+    android:id="@+id/filtersRoot"
+    style="@style/TrackFilters.Filters"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    tools:ignore="KeyboardInaccessibleWidget">
+
+    <TextView
+      android:id="@+id/title"
+      style="@style/TrackFilters.Filters.Title"
+      android:layout_width="@dimen/match_constraint"
+      android:layout_height="wrap_content"
+      android:layout_marginTop="@dimen/track_filters_title_margin_top"
+      android:layout_marginStart="@dimen/track_filters_title_margin_horizontal"
+      android:layout_marginEnd="@dimen/track_filters_title_margin_horizontal"
+      app:layout_constraintTop_toTopOf="parent"
+      app:layout_constraintStart_toStartOf="parent"
+      app:layout_constraintEnd_toStartOf="@+id/closeButton" />
+
+    <TextView
+      android:id="@+id/subtitle"
+      style="@style/TrackFilters.Filters.Subtitle"
+      android:layout_width="@dimen/match_constraint"
+      android:layout_height="wrap_content"
+      android:layout_marginTop="@dimen/track_filters_subtitle_margin_top"
+      app:layout_constraintStart_toStartOf="@+id/title"
+      app:layout_constraintTop_toBottomOf="@+id/title"
+      app:layout_constraintEnd_toEndOf="@+id/title" />
+
+    <ImageButton
+      android:id="@+id/closeButton"
+      style="@style/TrackFilters.Filters.CloseButton"
+      android:layout_width="@dimen/track_filters_close_button_size"
+      android:layout_height="@dimen/track_filters_close_button_size"
+      android:layout_marginTop="@dimen/track_filters_close_button_margin"
+      app:layout_constraintEnd_toEndOf="parent"
+      app:layout_constraintTop_toTopOf="parent" />
+
+    <android.support.v7.widget.RecyclerView
+      android:id="@+id/trackFiltersList"
+      style="@style/TrackFilters.Filters.Tracks"
+      android:layout_width="match_parent"
```

Comment:
Although it does work, `match_parent` in a `ConstraintLayout` child is not really ideal. You can achieve the same effect with `@dimen/match_constraint` and setting 
```
app:layout_constraintStart_toStartOf="parent"
app:layout_constraintEnd_toEndOf="parent"
```

Reply:
I know I know I'm just lazy XD

Reply:
223814812a7618cd17c3dd8318f825413a6b8702